base/src/com/thoughtworks/go/util/ZipUtil.java

```diff
@@ -128,6 +129,24 @@ public void unzip(ZipInputStream zipInputStream, File destDir) throws IOExceptio
         IOUtils.closeQuietly(zipInputStream);
     }
 
+    public String readContentFromFileInsideZip(ZipInputStream zipInputStream, String filePath) throws IOException {
+        File tempFolder = FileUtil.createTempFolder();
+        try {
+            ZipEntry zipEntry = zipInputStream.getNextEntry();
+            while (zipEntry != null) {
+                if (zipEntry.getName().equalsIgnoreCase(filePath)) {
+                    extractTo(zipEntry, zipInputStream, tempFolder);
+                    return FileUtil.readContentFromFile(new File(tempFolder, zipEntry.getName()));
+                }
+                zipEntry = zipInputStream.getNextEntry();
+            }
+        } finally {
+            IOUtils.closeQuietly(zipInputStream);
+            FileUtils.deleteQuietly(tempFolder);
```

Comment:
See https://stackoverflow.com/questions/15667125/read-content-from-files-which-are-inside-zip-file for doing this without an intermediate tempfile..